platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -0,0 +1,549 @@
+
+A new method will be added to the `PlatformBuilder` that will be used by the App to set the Genesis Roster.
+In most cases, there will be no need to call the method on `PlatformBuilder` to set the Genesis Roster except in the
+events we need to bootstrap a Genesis Network.
+
+```java
+//in PlatformBuilder
+
+/*
+  Set the Genesis Roster for the network.
+  <p>
+  This method is used to set the optional Genesis Roster for the network.
+  The Genesis Roster is a special roster used for the sole-purpose of bootstrapping a network.
+  The Genesis Roster, if set, is immutable and will be used to start a Genesis network Process or Network Transplant process.
+  <p>
+  @param genesisRoster the Genesis Roster for the network.
+ */
+void withGenesisRoster(@NonNull final Roster genesisRoster);
+```
+
+The logic to determine whether to invoke this method could be determined by some mechanism Services deems appropriate.
+
+#### Override Roster
+
+An `Override Roster` is an optional Roster that DevOps may provide for the explicit purpose of starting a network in a
+specific state,
+with a specific set of nodes, such as during a Network Transplant Process. A common example of this is Mainnet State
+being transplanted into Testnet.
+Provided an existing State and an `Override Roster`, the platform will adopt that roster.
+The next round number for this network will continue from the last round number in the provided state.
+The trigger for adopting the `Override Roster` will be the presence of an Override Roster at start-up with a State
+existing on disk.
+Bear in mind the Network Transplant process is designed primarily with test networks in mind.
+Upon adoption of the Override Roster, the Override Roster will be deleted from disk.
+In test networks where this process is designed to be used, the Override Roster can be easily accessed and deleted by
+DevOps.
+
+A new method will be added to the `PlatformBuilder` that will be used by the App to set the Override Roster.
+In most cases, as it's the case for a Genesis Roster, there will be no need to call the method on `PlatformBuilder`
+to set the Override Roster except in the events we need to transplant a network.
+
+```java
+//in PlatformBuilder
+/*
+  Set the Override Roster for the network.
+  <p>
+  This method is used to set the optional Override Roster for the network.
+  The Override Roster is a special roster used for the sole purpose of Transplanting an existing network.
+  The Override Roster, if set, is immutable and will be used to trigger a Network Transplant process.
+  <p>
+  @param overrideRoster the Override Roster for the network.
+ */
+void withOverrideRoster(@NonNull final Roster overrideRoster);
+```
+
+#### New Transplant Procedure
+
+The App will decide a network transplant sequence based on the following heuristics:
+
+Network Transplant process == Override Roster AND State provided
```

Comment:
I think this process will require a software upgrade as well. There will be events on disk in the PCES that are signed with the previous roster which need to be read into memory and have their signatures validated using the old roster. They way we identify the roster to use to verify PCES signatures is the event version. If the PCES event version is less than the current software version, the old roster is used. If the PCES event version is the same as the current software version, the active roster is used.

Reply:
updated.